src/Microsoft.ML.OnnxTransform/OnnxTransform.cs

```diff
@@ -0,0 +1,353 @@
+                    return getters;
+                }
+            }
+
+            private Delegate MakeGetter<T>(IRow input)
+            {
+                _host.AssertValue(input);
+                _host.Assert(typeof(T) == _outputItemRawType);
+
+                ValueGetter<VBuffer<T>> valuegetter = (ref VBuffer<T> dst) =>
+                {
+                    _idvToTensorAdapter.InitializeValueGetters(input);
+                    var inputTensors = new List<Tensor> { _idvToTensorAdapter.GetTensor() };
+                    var outputTensors = _parent.Model.Run(inputTensors);
+                    Contracts.Assert(outputTensors.Count() > 0);
+
+                    var values = dst.Values;
+                    if (Utils.Size(values) < _outputColType.VectorSize)
+                        values = new T[_outputColType.VectorSize];
+
+                    OnnxUtils.CopyTo(outputTensors[0], values);
+                    dst = new VBuffer<T>(values.Length, values, dst.Indices);
+                };
+
+                return valuegetter;
+            }
+        }
+    }
+    public sealed class OnnxEstimator : TrivialEstimator<OnnxTransform>
+    {
+        public OnnxEstimator(IHostEnvironment env, string modelFile, string input, string output)
+           : this(env, new OnnxTransform(env, modelFile, input, output))
+        {
+        }
+
+        public OnnxEstimator(IHostEnvironment env, OnnxTransform transformer)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(OnnxTransform)), transformer)
+        {
+        }
+
+        public override SchemaShape GetOutputSchema(SchemaShape inputSchema)
+        {
+            Host.CheckValue(inputSchema, nameof(inputSchema));
+            var result = inputSchema.Columns.ToDictionary(x => x.Name);
+            var resultDic = inputSchema.Columns.ToDictionary(x => x.Name);
+
+            var input = Transformer.Input;
+            if (!inputSchema.TryFindColumn(input, out var col))
+                throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", input);
+            if (!(col.Kind == SchemaShape.Column.VectorKind.VariableVector || col.Kind == SchemaShape.Column.VectorKind.Vector))
+                throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", input, nameof(VectorType), col.GetTypeString());
+            var inputNodeInfo = Transformer.Model.GetInputsInfo().Where(x => x.Name == input).First();
+            var expectedType = OnnxUtils.OnnxToMlNetType(inputNodeInfo.Type);
+            if (col.ItemType != expectedType)
+                throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", input, expectedType.ToString(), col.ItemType.ToString());
+
+            resultDic[Transformer.Output] = new SchemaShape.Column(Transformer.Output,
+                Transformer.OutputType.IsKnownSizeVector ? SchemaShape.Column.VectorKind.Vector
+                : SchemaShape.Column.VectorKind.VariableVector, Transformer.OutputType.ItemType, false);
```

Comment:
>Transformer.OutputType.ItemType [](start = 64, length = 31)

Either modify OnnxUtils.CopyTo to support other types rather than float, or put NumberType.R4 here.

Reply:
Done. 

---
In reply to: [220400133](https://github.com/dotnet/machinelearning/pull/942#discussion_r220400133) [](ancestors = 220400133)